adapters/smaato/smaato.go

```diff
@@ -260,20 +266,12 @@ func (adapter *adapter) makeRequest(request *openrtb2.BidRequest) (*adapters.Req
 	}, nil
 }
 
-func getAdMarkupType(response *adapters.ResponseData, adMarkup string) (adMarkupType, error) {
+func getAdMarkupType(response *adapters.ResponseData) (adMarkupType, error) {
 	if admType := adMarkupType(response.Headers.Get("X-Smt-Adtype")); admType != "" {
 		return admType, nil
-	} else if strings.HasPrefix(adMarkup, `{"image":`) {
-		return smtAdTypeImg, nil
-	} else if strings.HasPrefix(adMarkup, `{"richmedia":`) {
-		return smtAdTypeRichmedia, nil
-	} else if strings.HasPrefix(adMarkup, `<?xml`) {
-		return smtAdTypeVideo, nil
-	} else if strings.HasPrefix(adMarkup, `{"native":`) {
-		return smtAdTypeNative, nil
 	} else {
 		return "", &errortypes.BadServerResponse{
-			Message: fmt.Sprintf("Invalid ad markup %s.", adMarkup),
+			Message: fmt.Sprintf("X-Smt-Adtype header is missing!"),
```

Comment:
Nitpick: Consider changing the `!` to `.`. We keep to plain punctuation for errors.

Reply:
fixed